### ISSUE TITLE
Error when saving a dropdown inside a table field

### ISSUE THREAD

**Issue by smcyr** [verbb/formie#1172]
### Describe the bug

When creating a new form with a dropdown inside a table field (with some options), there is this error when saving the form:

```
error: "Setting unknown property: verbb\\formie\\fields\\formfields\\Table::form_2"
```

It seems that the options of the dropdown are duplicated inside the property "form_X" in "settings" in addition to the "column". 

I was able to fix the issue with this quickfix inside the `forms.php` service at line 484:
```
foreach ($settings as $key => $value) {
  if (str_starts_with($key, "form_")) {
    ArrayHelper::remove($settings, $key);
  }
}
```

But it should propably be fixed at the Javascript level.

Here is the payload:

```
[
    {
        "id": 4697,
        "label": "Page 1",
        "rows": [
            {
                "id": "new2977-6888",
                "fields": [
                    {
                        "id": 1343,
                        "name": "Test",
                        "handle": "test",
                        "columnSuffix": "gtdgcuzo",
                        "label": "Test",
                        "type": "verbb\\formie\\fields\\formfields\\Table",
                        "hasLabel": true,
                        "settings": {
                            "searchable": true,
                            "columns": [
                                {
                                    "heading": "Test",
                                    "handle": "test",
                                    "width": "",
                                    "type": "singleline",
                                    "id": "col1"
                                },
                                {
                                    "heading": "Test 2",
                                    "handle": "test2",
                                    "width": "",
                                    "type": "select",
                                    "options": [
                                        {
                                            "label": "Test",
                                            "value": "Test"
                                        },
                                        {
                                            "label": "Test2",
                                            "value": "Test2"
                                        }
                                    ],
                                    "id": "col2"
                                }
                            ],
                            "static": false,
                            "addRowLabel": "Add row",
                            "maxRows": null,
                            "minRows": null,
                            "defaults": [
                                []
                            ],
                            "columnType": "text",
                            "matchField": null,
                            "placeholder": null,
                            "defaultValue": null,
                            "prePopulate": null,
                            "errorMessage": null,
                            "labelPosition": null,
                            "instructionsPosition": null,
                            "cssClasses": null,
                            "containerAttributes": null,
                            "inputAttributes": null,
                            "includeInEmail": true,
                            "enableConditions": false,
                            "conditions": null,
                            "enableContentEncryption": false,
                            "visibility": null,
                            "formId": 160387,
                            "isNested": false,
                            "label": "Test",
                            "handle": "test",
                            "required": false,
                            "instructions": null,
                            "form_2": {
                                "heading": "Test 2",
                                "handle": "test2",
                                "width": "",
                                "type": "select",
                                "options": [
                                    {
                                        "label": "Test",
                                        "value": "Test"
                                    },
                                    {
                                        "label": "Test2",
                                        "value": "Test2"
                                    }
                                ],
                                "id": "col2"
                            }
                        },
                        "isCosmetic": false,
                        "isSynced": false,
                        "hasConditions": false,
                        "supportsNested": false,
                        "isNested": false,
                        "vid": "new7773-3842",
                        "brandNewField": false
                    }
                ]
            }
        ],
        "settings": {
            "submitButtonLabel": "Submit",
            "backButtonLabel": "Back",
            "showBackButton": false,
            "saveButtonLabel": "Save",
            "showSaveButton": false,
            "saveButtonStyle": "link",
            "buttonsPosition": "left",
            "cssClasses": null,
            "containerAttributes": null,
            "inputAttributes": null,
            "enableNextButtonConditions": false,
            "nextButtonConditions": [],
            "enablePageConditions": false,
            "pageConditions": [],
            "enableJsEvents": false,
            "jsGtmEventOptions": []
        }
    }
]
```

Here is the stack trace:
```
[
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\BaseYii.php",
    "line": 558,
    "function": "__set",
    "class": "yii\\base\\Component",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\base\\BaseObject.php",
    "line": 107,
    "function": "configure",
    "class": "yii\\BaseYii",
    "type": "::"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\base\\Model.php",
    "line": 78,
    "function": "__construct",
    "class": "yii\\base\\BaseObject",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\fields\\Table.php",
    "line": 146,
    "function": "__construct",
    "class": "craft\\base\\Model",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\verbb\\formie\\src\\base\\FormFieldTrait.php",
    "line": 128,
    "function": "__construct",
    "class": "craft\\fields\\Table",
    "type": "->"
  },
  {
    "function": "__construct",
    "class": "verbb\\formie\\fields\\formfields\\Table",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\di\\Container.php",
    "line": 419,
    "function": "newInstanceArgs",
    "class": "ReflectionClass",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\di\\Container.php",
    "line": 170,
    "function": "build",
    "class": "yii\\di\\Container",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\BaseYii.php",
    "line": 365,
    "function": "get",
    "class": "yii\\di\\Container",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\Craft.php",
    "line": 53,
    "function": "createObject",
    "class": "yii\\BaseYii",
    "type": "::"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\helpers\\Component.php",
    "line": 122,
    "function": "createObject",
    "class": "Craft",
    "type": "::"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\services\\Fields.php",
    "line": 593,
    "function": "createComponent",
    "class": "craft\\helpers\\Component",
    "type": "::"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\verbb\\formie\\src\\services\\Forms.php",
    "line": 507,
    "function": "createField",
    "class": "craft\\services\\Fields",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\verbb\\formie\\src\\services\\Forms.php",
    "line": 357,
    "function": "buildFieldLayout",
    "class": "verbb\\formie\\services\\Forms",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\verbb\\formie\\src\\services\\Forms.php",
    "line": 444,
    "function": "assembleLayout",
    "class": "verbb\\formie\\services\\Forms",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\verbb\\formie\\src\\controllers\\FormsController.php",
    "line": 145,
    "function": "buildFormFromPost",
    "class": "verbb\\formie\\services\\Forms",
    "type": "->"
  },
  {
    "function": "actionSave",
    "class": "verbb\\formie\\controllers\\FormsController",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\base\\InlineAction.php",
    "line": 57,
    "function": "call_user_func_array"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\base\\Controller.php",
    "line": 178,
    "function": "runWithParams",
    "class": "yii\\base\\InlineAction",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\base\\Module.php",
    "line": 552,
    "function": "runAction",
    "class": "yii\\base\\Controller",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\web\\Application.php",
    "line": 301,
    "function": "runAction",
    "class": "yii\\base\\Module",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\web\\Application.php",
    "line": 625,
    "function": "runAction",
    "class": "craft\\web\\Application",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\craftcms\\cms\\src\\web\\Application.php",
    "line": 280,
    "function": "_processActionRequest",
    "class": "craft\\web\\Application",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\vendor\\yiisoft\\yii2\\base\\Application.php",
    "line": 384,
    "function": "handleRequest",
    "class": "craft\\web\\Application",
    "type": "->"
  },
  {
    "file": "C:\\wamp64\\www\\web\\index.php",
    "line": 12,
    "function": "run",
    "class": "yii\\base\\Application",
    "type": "->"
  }
]
```

### Steps to reproduce

1. Create new form 
2. Create table field
3. Create a dropdown field within the table field
4. Add some options to the dropdown field
5. Save the field

### Form settings

- Multi-page form: No

### Craft CMS version

4.3.1

### Plugin version

2.0.17

### Multi-site?

Yesy

**Comment by engram-design** [comment]
Hmmm, I can't seem to replicate this error, and I've followed your steps exactly.

Are you able to have a look at the payload being sent to the server? It should look something like: 
![image](https://user-images.githubusercontent.com/1221575/201800906-67257818-42e3-4e6c-b3f8-26d930289ea8.png)

(if you can respond with text actual text, not a screenshot so I can format it).

**Comment by smcyr** [comment]
I tried in a new setup and still the same error. Here is the payload:

```
[{"id":18,"label":"Page 1","rows":[{"id":"new6291-1250","fields":[{"vid":"new1512-3278","type":"verbb\\formie\\fields\\formfields\\Table","columnWidth":12,"hasLabel":true,"settings":{"labelPosition":"","instructionsPosition":"","addRowLabel":"Add row","label":"Test table","handle":"testTable","columns":[{"heading":"Test dropdown","handle":"testDropdown","width":"","type":"select","id":"col1","options":[{"label":"Test dropdown option","value":"Test dropdown option"}]}],"visibility":"","form_2":{"heading":"Test dropdown","handle":"testDropdown","width":"","type":"select","id":"col1","options":[{"label":"Test dropdown option","value":"Test dropdown option"}]}},"brandNewField":false,"label":"Test table","handle":"testTable"}]}],"settings":{"submitButtonLabel":"Submit","backButtonLabel":"Back","showBackButton":false,"saveButtonLabel":"Save","showSaveButton":false,"saveButtonStyle":"link","buttonsPosition":"left","cssClasses":null,"containerAttributes":null,"inputAttributes":null,"enableNextButtonConditions":false,"nextButtonConditions":[],"enablePageConditions":false,"pageConditions":[],"enableJsEvents":false,"jsGtmEventOptions":[]}}]
```

And a screenshot:
![Screenshot 2022-11-15 083049](https://user-images.githubusercontent.com/664920/201931909-766bdf36-c68d-4b47-8039-e3cab4068c33.png)

Here is some additional info:

Application Info
PHP version	8.0.25
OS version	WINNT 10.0
Database driver & version	MySQL 10.6.9
Image driver & version	GD 8.0.25
Craft edition & version	Craft Pro 4.3.1
Yii version	2.0.46
Twig version	v3.4.3
Guzzle version	7.5.0

Plugins
Amazon S3	2.0.1
Formie	2.0.17
Redactor	3.0.2
Translations	3.0.1
Vite	4.0.3

Modules
website	modules\website\WebsiteModule
verbb-base	verbb\base\Base
debug	craft\debug\Module

**Comment by engram-design** [comment]
Thanks for that, were there are other console errors worth noting? I can see there are 3, but they might just be related to the failed form save, but wanted to check if there's any JS errors.

**Comment by smcyr** [comment]
Here are the console errors:

```
axios.js?v=1667221712:2 
        
       POST http://craft.test/index.php?p=admin%2Factions%2Fformie%2Fforms%2Fsave&site=english&v=1668630674261 500 (Internal Server Error)
(anonymous) @ axios.js?v=1667221712:2
e.exports @ axios.js?v=1667221712:2
e.exports @ axios.js?v=1667221712:2
f.request @ axios.js?v=1667221712:2
(anonymous) @ axios.js?v=1667221712:2
(anonymous) @ Craft.js:816
sendActionRequest @ Craft.js:801
onSave @ formie-form.js:243
(anonymous) @ formie-form.js:122
(anonymous) @ mitt.es.js:57
emit @ mitt.es.js:57
(anonymous) @ formie-form.js:464
Promise.then (async)
Hr @ runtime-core.esm-bundler.js:246
onSave @ formie-form.js:463
(anonymous) @ runtime-dom.esm-bundler.js:1481
Wr @ runtime-core.esm-bundler.js:155
jn @ runtime-core.esm-bundler.js:164
n @ runtime-dom.esm-bundler.js:369
formie-form.js:273 
        
       o {message: 'Request failed with status code 500', name: 'AxiosError', code: 'ERR_BAD_RESPONSE', config: {…}, request: XMLHttpRequest, …}
(anonymous) @ formie-form.js:273
Promise.catch (async)
onSave @ formie-form.js:272
(anonymous) @ formie-form.js:122
(anonymous) @ mitt.es.js:57
emit @ mitt.es.js:57
(anonymous) @ formie-form.js:464
Promise.then (async)
Hr @ runtime-core.esm-bundler.js:246
onSave @ formie-form.js:463
(anonymous) @ runtime-dom.esm-bundler.js:1481
Wr @ runtime-core.esm-bundler.js:155
jn @ runtime-core.esm-bundler.js:164
n @ runtime-dom.esm-bundler.js:369
formie-form.js:320 
        
       Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forEach')
    at Proxy.onError (formie-form.js:320:43)
    at formie-form.js:275:30
```

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.17",
  "...": "..."
}
```

Then run `composer update`.

**Comment by smcyr** [comment]
Thanks, I did the update and it works now!

**Comment by engram-design** [comment]
Fixed in [2.0.18](https://github.com/verbb/formie/releases/tag/2.0.18)
